src/app/vertextool/qgsvertextool.cpp

```diff
@@ -849,6 +841,126 @@ QgsPointLocator::Match QgsVertexTool::snapToPolygonInterior( QgsMapMouseEvent *e
 }
 
 
+QList<QgsPointLocator::Match> QgsVertexTool::findEditableLayerMatches( const QgsPointXY &mapPoint, QgsVectorLayer *layer )
+{
+  QgsPointLocator::MatchList matchList;
+
+  if ( !layer->isEditable() )
+    return matchList;
+
+  QgsSnappingUtils *snapUtils = canvas()->snappingUtils();
+  QgsPointLocator *locator = snapUtils->locatorForLayer( layer );
+
+  if ( layer->geometryType() == QgsWkbTypes::PolygonGeometry )
+  {
+    matchList << locator->pointInPolygon( mapPoint );
+  }
+
+  double tolerance = QgsTolerance::vertexSearchRadius( canvas()->mapSettings() );
+  matchList << locator->edgesInRect( mapPoint, tolerance );
+  matchList << locator->verticesInRect( mapPoint, tolerance );
+
+  return matchList;
+}
+
+
+QSet<QPair<QgsVectorLayer *, QgsFeatureId> > QgsVertexTool::findAllEditableFeatures( const QgsPointXY &mapPoint )
+{
+  QSet< QPair<QgsVectorLayer *, QgsFeatureId> > alternatives;
+
+  // if there is a current layer, it should have priority over other layers
+  // because sometimes there may be match from multiple layers at one location
+  // and selecting current layer is an easy way for the user to prioritize a layer
+  if ( QgsVectorLayer *currentVlayer = currentVectorLayer() )
+  {
+    for ( const QgsPointLocator::Match &m : findEditableLayerMatches( mapPoint, currentVlayer ) )
```

Comment:
could you do ```const auto &m```

Reply:
There was discussion on this when we first moved to C++11, but the outcome was to stick to explicit types UNLESS they are iterators or over complex template based types. (Avoiding auto improves readability)

Reply:
That's a fair point, thanks for the info!